mm/slab_common.c

```diff
@@ -449,9 +450,6 @@ static int shutdown_cache(struct kmem_cache *s)
 	/* free asan quarantined objects */
 	kasan_cache_shutdown(s);
 
-	if (!kfence_shutdown_cache(s))
```

Comment:
Ah, ok, that is effectively what you are doing - moving this call after __kmem_cache_shutdown, so that we'll leak the whole cache in the case of an early exit from shutdown_cache(), correct?

Reply:
Yes, if there are non-KFENCE objects there's nothing we can do. But if there are only KFENCE objects, we'll let the kernel release the cache normally, we just leak some KFENCE objects by zombie'fying them. We could maybe do better to provide helpful error messages by calling kfence_shutdown_cache() in all cases (including early return due to __kmem_cache_shutdown()), but with the whole SLAB_TYPESAFE_BY_RCU business this becomes really ugly and is way more than the 2 calls to kfence_shutdown_cache() we have now.

The current version is safe for RCU and non-RCU cases, but maybe we won't get very helpful error reports if there are other non-KFENCE objects still allocated.